src/avro_json_decoder.erl

```diff
@@ -511,8 +511,14 @@ parse_union(_, _, _, _, _) ->
   erlang:error(wrong_union_value).
 
 %% @private
-parse_union_ex(ValueTypeName, Value, UnionType,
-               ExtractFun, IsWrapped, Hook) ->
+parse_union_ex(ValueTypeName, Value, UnionType, ExtractFun, IsWrapped, Hook) ->
+  Hook(UnionType, ValueTypeName, Value,
+       fun(In) -> do_parse_union_ex(ValueTypeName, In, UnionType,
```

Comment:
Sorry for being too demanding, but setting `end` on the same level as `fun` is more clear:

    fun(In) -> 
       do_parse_union_ex(ValueTypeName, In, UnionType,
                                    ExtractFun, IsWrapped, Hook) 
    end).